### PR TITLE
[MM-42400] Prevent making groups with reserved names

### PR DESCRIPTION
**Pull request by cyrilzhang-mm**
#### Summary

Prevent making user groups with reserved names in the API.

#### Ticket Link

[Jira ticket](https://mattermost.atlassian.net/browse/MM-42400)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: model/group.go [hunk 1]
```diff
@@ -218,6 +218,10 @@ func (group *Group) IsValidName() *AppError {
 			return NewAppError("Group.IsValidName", "model.group.name.invalid_length.app_error", map[string]any{"GroupNameMaxLength": GroupNameMaxLength}, "", http.StatusBadRequest)
 		}
 
+		if *group.Name == UserNotifyAll || *group.Name == ChannelMentionsNotifyProp || *group.Name == UserNotifyHere {
```
Comment (Cijin) [inline model/group.go:221]: Does this need to be checked when group name is updated?

Comment (cyrilzhang-mm) [reply model/group.go:221]: This function is also called, by `IsValidForUpdate`, when the group name is updated, so it is checked.

**Review by BenCookie95 — CHANGES_REQUESTED** [review]
Can we add some tests that try to create/update a group with the reserved names?

**Review by plant99 — APPROVED** [review]
LGTM.

**Review by BenCookie95 — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Working as expected.

**Before**
image

**After**

image

**Review by Cijin — APPROVED** [review]

**Pull request merged**